### ISSUE TITLE
Authentication failed...

### ISSUE THREAD

**Issue by kstar0101** [RainLoop/rainloop-webmail#2085]
**RainLoop version: 1.15.0, browser: chrome, OS:centos7**

I have completed all settings including domain in the admin screen, but I can't log in to rainloop.
My email address and password is correct and I tried changing all the configurations(secure mode, Use short login and port).
The following is the log file.
`
![Screenshot_5](https://user-images.githubusercontent.com/53885804/113530718-a7d7bf00-9601-11eb-85a8-760d937c3e8a.png)

[02:15:46.064][92e42f0c] INFO[DATA]: [DATE:05.04.21][OFFSET:-00][RL:1.15.0][PHP:7.4.16][IP:188.43.136.32][PID:32268][nginx/1.16.1][fpm-fcgi]
[02:15:46.064][92e42f0c] INFO[DATA]: [APC:off][MB:on][PDO:mysql,sqlite][Streams:tcp,udp,unix,udg,ssl,sslv3,tls,tlsv1.0,tlsv1.1,tlsv1.2]
[02:15:46.064][92e42f0c] REQUEST[NOTE]: [GET] https://contact.sai-katsu.com/
[02:15:46.078][92e42f0c] INFO[MEMORY]: Memory peak usage: 2MB
[02:15:46.078][92e42f0c] INFO[TIME]: Time delta: 0.043928861618042

[02:15:47.048][588a2f64] INFO[DATA]: [DATE:05.04.21][OFFSET:-00][RL:1.15.0][PHP:7.4.16][IP:188.43.136.32][PID:21647][nginx/1.16.1][fpm-fcgi]
[02:15:47.048][588a2f64] INFO[DATA]: [APC:off][MB:on][PDO:mysql,sqlite][Streams:tcp,udp,unix,udg,ssl,sslv3,tls,tlsv1.0,tlsv1.1,tlsv1.2]
[02:15:47.048][588a2f64] REQUEST[NOTE]: [GET] https://contact.sai-katsu.com/?/Css/0/User/-/SquaresDark/-/c2bffd407d3244610dbc47fb8cd3dd29/Hash/-/
[02:15:47.057][588a2f64] INFO[MEMORY]: Memory peak usage: 2MB
[02:15:47.057][588a2f64] INFO[TIME]: Time delta: 0.038181066513062

[02:15:48.921][813cf673] INFO[DATA]: [DATE:05.04.21][OFFSET:-00][RL:1.15.0][PHP:7.4.16][IP:188.43.136.32][PID:32264][nginx/1.16.1][fpm-fcgi]
[02:15:48.921][813cf673] INFO[DATA]: [APC:off][MB:on][PDO:mysql,sqlite][Streams:tcp,udp,unix,udg,ssl,sslv3,tls,tlsv1.0,tlsv1.1,tlsv1.2]
[02:15:48.921][813cf673] REQUEST[NOTE]: [GET] https://contact.sai-katsu.com/?/AppData@no-mobile-0/0/4012818613074971/
[02:15:48.925][813cf673] APPDATA[DATA]: if(window.__initAppData){window.__initAppData({"Auth":false,"AccountHash":"","AccountSignMe":false,"AuthAccountHash":"","MailToEmail":"","Email":"","DevEmail":"","DevPassword":"","Title":"contact.sai-katsu.com","LoadingDescription":"RainLoop","LoadingDescriptionEsc":"RainLoop","FaviconUrl":"","LoginDescription":"","LoginLogo":"","LoginBackground":"","LoginCss":"","UserLogo":"","UserLogoTitle":"","UserLogoMessage":"","UserCss":"","WelcomePageUrl":"","WelcomePageDisplay":"","IncludeCss":"","IncludeBackground":"","LoginDefaultDomain":"sai-katsu.com","DetermineUserLanguage":true,"DetermineUserDomain":true,"UseLoginWelcomePage":false,"StartupUrl":"","SieveAllowFileintoInbox":false,"ContactsIsAllowed":false,"ChangePasswordIsAllowed":false,"RequireTwoFactor":false,"Community":false,"PremType":false,"Admin":[],"Capa":["FOLDERS","MESSAGELIST_ACTIONS","DANGEROUS_ACTIONS","RELOAD","QUOTA","SETTINGS","ADDITIONAL_ACCOUNTS","IDENTITIES","THEMES","FILTERS","HELP","ATTACHMENTS_ACTIONS","MESSAGE_ACTIONS","COMPOSER","CONTACTS","SEARCH","SEARCH_ADV","GRAVATAR","ATTACHMENT_THUMBNAILS","PREFETCH","AUTOLOGOUT"],"Plugins":[],"System":{"version":"1.15.0","admin":false,"mobile":false,"mobileDevice":false,"webPath":"","webVersionPath":"rainloop\/v\/1.15.0\/","token":"16e0d30c7a00f6f9487805fbc778cc77","inIframe":false,"allowHtmlEditorSourceButton":false,"allowHtmlEditorBitiButtons":false,"allowCtrlEnterOnCompose":true,"customLoginLink":"","customLogoutLink":"","forgotPasswordLinkUrl":"","registrationLinkUrl":"","hideSubmitButton":true,"jsHash":"afac1a51fc7cde30802c4f156f721dc6","useImapThread":true,"useImapSubscribe":true,"allowAppendMessage":false,"materialDesign":true,"folderSpecLimit":50,"faviconStatus":true,"allowCmdInterface":false,"useNativeScrollbars":false,"listPermanentFiltered":false,"themes":["Default","A","Blurred","DarkShine","Linen","Love","Snow","Squares","SquaresDark","Stripes","Wood","Xv","Clear"],"languages":["ar_SA","bg_BG","cs_CZ","da_DK","de_DE","el_GR","en_GB","en_US","es_ES","et_EE","fa_IR","fi_FI","fr_FR","hu_HU","id_ID","is_IS","it_IT","ja_JP","ko_KR","lt_LT","lv_LV","nb_NO","nl_NL","pl_PL","pt_BR","pt_PT","ro_RO","ru_RU","sk_SK","sl_SI","sv_SE","tr_TR","uk_UA","zh_CN","zh_TW"],"languagesAdmin":["cs_CZ","da_DK","de_DE","en_US","es_ES","fa_IR","fi_FI","fr_FR","hu_HU","id_ID","it_IT","ja_JP","lt_LT","nb_NO","nl_NL","pl_PL","pt_BR","ru_RU","sk_SK","sl_SI","sv_SE","zh_CN"],"appVersionType":"standard","attachmentsActions":["zip"]},"AdditionalLoginError":"","AllowGoogleSocial":false,"AllowGoogleSocialAuth":false,"AllowGoogleSocialAuthGmail":false,"AllowGoogleSocialDrive":false,"AllowGoogleSocialPreview":false,"GoogleClientID":"","GoogleApiKey":"","AllowFacebookSocial":false,"AllowTwitterSocial":false,"AllowDropboxSocial":false,"DropboxApiKey":"","SupportedFacebookSocial":true,"ProjectHash":"c2bffd407d3244610dbc47fb8cd3dd29","NewMoveToFolder":true,"AllowLanguagesOnSettings":true,"AllowLanguagesOnLogin":true,"AttachmentLimit":26214400,"SignMe":"DefaultOff","UseLocalProxyForExternalImages":false,"ShowImages":false,"MPP":20,"SoundNotification":false,"DesktopNotifications":false,"Layout":1,"EditorDefaultType":"Html","UseCheckboxesInList":true,"AutoLogout":30,"UseThreads":false,"AllowDraftAutosave":true,"ReplySameFolder":false,"ContactsAutosave":true,"EnableTwoFactor":false,"ParentEmail":"","InterfaceAnimation":true,"UserBackgroundName":"","UserBackgroundHash":"","Theme":"SquaresDark","NewThemeLink":".\/?\/Css\/0\/User\/-\/SquaresDark\/-\/c2bffd407d3244610dbc47fb8cd3dd29\/Hash\/-\/","Language":"en_US","LanguageAdmin":"en_US","UserLanguageRaw":"en_US","UserLanguage":"en_US","UserLanguageAdmin":"en_US","PluginsLink":"","LangLink":".\/?\/Lang\/0\/App\/en_US\/c2bffd407d3244610dbc47fb8cd3dd29\/","StaticLibJsLink":"rainloop\/v\/1.15.0\/static\/js\/min\/libs.min.js?standard","StaticAppJsLink":"rainloop\/v\/1.15.0\/static\/js\/min\/app.min.js?standard","StaticAppJsNextLink":"rainloop\/v\/1.15.0\/static\/js\/app.next.js?standard","StaticEditorJsLink":"rainloop\/v\/1.15.0\/static\/ckeditor\/ckeditor.js?standard"});}
[02:15:48.933][813cf673] INFO[MEMORY]: Memory peak usage: 2MB
[02:15:48.933][813cf673] INFO[TIME]: Time delta: 0.040858030319214

[02:15:49.401][ca739ab4] INFO[DATA]: [DATE:05.04.21][OFFSET:-00][RL:1.15.0][PHP:7.4.16][IP:188.43.136.32][PID:404][nginx/1.16.1][fpm-fcgi]
[02:15:49.401][ca739ab4] INFO[DATA]: [APC:off][MB:on][PDO:mysql,sqlite][Streams:tcp,udp,unix,udg,ssl,sslv3,tls,tlsv1.0,tlsv1.1,tlsv1.2]
[02:15:49.401][ca739ab4] REQUEST[NOTE]: [GET] https://contact.sai-katsu.com/?/Lang/0/App/en_US/c2bffd407d3244610dbc47fb8cd3dd29/
[02:15:49.411][ca739ab4] INFO[MEMORY]: Memory peak usage: 2MB
[02:15:49.411][ca739ab4] INFO[TIME]: Time delta: 0.03909707069397

[02:16:01.348][260dacff] INFO[DATA]: [DATE:05.04.21][OFFSET:-00][RL:1.15.0][PHP:7.4.16][IP:188.43.136.32][PID:32265][nginx/1.16.1][fpm-fcgi]
[02:16:01.348][260dacff] INFO[DATA]: [APC:off][MB:on][PDO:mysql,sqlite][Streams:tcp,udp,unix,udg,ssl,sslv3,tls,tlsv1.0,tlsv1.1,tlsv1.2]
[02:16:01.348][260dacff] REQUEST[NOTE]: [POST] https://contact.sai-katsu.com/?/Ajax/&q[]=/0/
[02:16:01.350][260dacff] AJAX[NOTE]: Action: DoLogin
[02:16:01.350][260dacff] POST[DATA]: {"Email":"canopus@sai-katsu.com","Login":"","Password":"*******","Language":"","AdditionalCode":"","AdditionalCodeSignMe":"0","SignMe":"0","Action":"Login","XToken":"16e0d30c7a00f6f9487805fbc778cc77"}
[02:16:01.358][260dacff] IMAP[NOTE]: Start connection to "tcp://imap.sai-katsu.conoha.io:143"
[02:16:01.367][260dacff] IMAP[NOTE]: Connected (success)
[02:16:01.367][260dacff] IMAP[TIME]: 0.0091390609741211 (raw connection)
[02:16:01.368][260dacff] IMAP[DATA]: < * OK [CAPABILITY IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE IDLE STARTTLS AUTH=PLAIN AUTH=LOGIN] Dovecot ready.\r\n
[02:16:01.368][260dacff] IMAP[TIME]: 0.0099809169769287 (*)
[02:16:01.368][260dacff] IMAP[DATA]: > TAG1 AUTHENTICATE PLAIN\r\n
[02:16:01.369][260dacff] IMAP[DATA]: < + \r\n
[02:16:01.369][260dacff] IMAP[TIME]: 0.0006859302520752 (TAG1)
[02:16:01.369][260dacff] IMAP[SECURE]: > *******\r\n
[02:16:03.382][260dacff] IMAP[DATA]: < TAG1 NO [AUTHENTICATIONFAILED] Authentication failed.\r\n
[02:16:03.382][260dacff] IMAP[WARNING]: MailSo\Imap\Exceptions\NegativeResponseException: MailSo-Imap-Exceptions-NegativeResponseException (ImapClient.php ~ 1876) in /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/MailSo/Imap/ImapClient.php:1876
Stack trace:
#0 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/MailSo/Imap/ImapClient.php(1953): MailSo\Imap\ImapClient->validateResponse()
#1 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/MailSo/Imap/ImapClient.php(283): MailSo\Imap\ImapClient->parseResponseWithValidation()
#2 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/MailSo/Mail/MailClient.php(93): MailSo\Imap\ImapClient->Login()
#3 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/Model/Account.php(508): MailSo\Mail\MailClient->Login()
#4 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/Actions.php(2153): RainLoop\Model\Account->IncConnectAndLoginHelper()
#5 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/Actions.php(2404): RainLoop\Actions->CheckMailConnection()
#6 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/Actions.php(2456): RainLoop\Actions->LoginProcess()
#7 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/ServiceActions.php(174): RainLoop\Actions->DoLogin()
#8 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/Service.php(150): RainLoop\ServiceActions->ServiceAjax()
#9 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/Service.php(62): RainLoop\Service->localHandle()
#10 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/libraries/RainLoop/Service.php(85): RainLoop\Service->__construct()
#11 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/app/handle.php(94): RainLoop\Service::Handle()
#12 /var/webdata/sites/contact.sai-katsu.com/html/rainloop/v/1.15.0/include.php(243): include('/var/webdata/si...')
#13 /var/webdata/sites/contact.sai-katsu.com/html/index.php(13): include('/var/webdata/si...')
#14 {main}
`
Please help me.

**Comment by kstar0101** [comment]
I debugged the code and found the error.
My mail server includes the domain in the user ID (eg admin@example.com).
However, rainloop automatically deleted the domain part when logging in.
Because of this, the identification information became wrong.

Accurate login information
user: admin@example.com
pass: password

Login information attempted by rainloop
user: admin
pass: password

I fixed the error by modifying file(rainloop/v*/app/libraries/RainLoop/Model/Account.php).
